hazelcast/src/main/java/com/hazelcast/replicatedmap/impl/ReplicatedMapProxy.java

```diff
@@ -88,6 +89,7 @@
     private static final int RETRY_INTERVAL_COUNT = 3;
     private static final int KEY_SET_MIN_SIZE = 16;
     private static final int KEY_SET_STORE_MULTIPLE = 4;
+    private static final int PARALLEL_INIT_REQUESTS_LIMIT = 100;
```

Comment:
I actually now see that the "async" loading will start loading all partitions in parallel while the "sync" one will throttle. I also see that the throttling is a bit broken in the sense that the limit is not applied if the for loop ends and we restart the while loop.

Nevermind, let's leave it as-is. Since async fillup does no throttling at all, the sync can work with the best-effort approach for now. We can always add it to the "async" loading later, add more robust throttling on both or change this number as well since it's all a private implementation detail.

Reply:
Regarding the reset limit in the while loop. I actually did it intentionally - the current (as well as the old logic) does not track statuses for the sync operations invocations. If a ReplicatedRecordStore for some partition is not loaded after a "timeout" (1s sleep) - we treat it as an issue and do a retry in the next round. So, the idea here is to limit the number of concurrent sync requests which we can send for each such round, like (not more than 100 sync requests per 1 second)...

Agree, that is not so good, ideally a rate limiting via tracking the number of real in-flight invocations would be better but it will require much more changes I suppose..

Reply:
👍